### DISCUSSION TITLE
Kubernetes NugGet Packaging

### DISCUSSION THREAD

**Discussion by dpbevin** [dotnet/yarp#1631] [Ideas] [answered]
Several people have expressed the desire to consume the Kubernetes YARP via NuGet packages. As of 3-April-2022, there are **seven** different C# projects (in this repo) that comprise the Kubernetes support for YARP.

## Proposal
I'm proposing that we collapse these seven projects into a single project before considering NuGet packaging for the Kubernetes support.

There may also be an opportunity to simplify the generalized "OperatorFramework" code as part of the process and make the code more specific to the job at hand.

@specialforest @MihaZupan

**Comment by karelz** [comment, marked as answer]
Let's track the work and final decision (yes/no) in #1640
